app/models/katello/glue/pulp/repos.rb

```diff
@@ -172,36 +172,34 @@ def latest_sync_statuses
           r._get_most_recent_sync_status
         end
       end
+    # Get the most relevant status for all the repos in this Product
+    def sync_status
+      return @status if @status
+      statuses = repos(self.library, nil, false).map {|r|  ::Katello::Presenters::SyncStatus.new(r).sync_progress}
 
-      # Get the most relevant status for all the repos in this Product
-      def sync_status
-        return @status if @status
+      return PulpSyncStatus.new(:state => PulpSyncStatus::Status::NOT_SYNCED) if statuses.empty?
 
-        statuses = repos(self.library, nil, false).map { |r| r.sync_status }
-        return PulpSyncStatus.new(:state => PulpSyncStatus::Status::NOT_SYNCED) if statuses.empty?
+      relevant_states = [::Katello::Presenters::SyncStatus::STATUS_VALUES[:running],
+                 ::Katello::Presenters::SyncStatus::STATUS_VALUES[:never_synced],
+                 ::Katello::Presenters::SyncStatus::STATUS_VALUES[:canceled],
+                 ::Katello::Presenters::SyncStatus::STATUS_VALUES[:error]]
 
-        #if any of repos sync still running -> product sync running
-        idx = statuses.index { |r| r.state.to_s == PulpSyncStatus::Status::RUNNING.to_s }
-        return statuses[idx] unless idx.nil?
+      #if any of repos sync still running -> product sync running
+      idx = statuses.index { |r| r[:is_running] }
+      return statuses[idx][:state] if !idx.nil?
 
-        #else if any of repos not synced -> product not synced
-        idx = statuses.index { |r| r.state.to_s == PulpSyncStatus::Status::NOT_SYNCED.to_s }
-        return statuses[idx] unless idx.nil?
-
-        #else if any of repos sync cancelled -> product sync cancelled
-        idx = statuses.index { |r| r.state.to_s == PulpSyncStatus::Status::CANCELED.to_s }
-        return statuses[idx] unless idx.nil?
-
-        #else if any of repos sync finished with error -> product sync finished with error
-        idx = statuses.index { |r| r.state.to_s == PulpSyncStatus::Status::ERROR.to_s }
-        return statuses[idx] unless idx.nil?
-
-        #else -> all finished
-        @status = statuses[0]
+      #else if any of repos not synced -> product not synced
+      idx = statuses.index do |s|
+        relevant_states.find(s[:state])
       end
+      return statuses[idx][:state] if !idx.nil?
+
+      #else -> all finished
+      @status = statuses[0]
+    end
 
       def sync_state
-        self.sync_status.state
+        self.sync_status[:state]
```

Comment:
it was coming from sync_status, not consistently returning the same type object.